src/main/java/com/ocient/cli/extract/ExtractConfiguration.java

```
@@ -0,0 +1,203 @@
+package com.ocient.cli.extract;
+
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+import org.apache.commons.collections4.IteratorUtils;
+import org.apache.commons.configuration2.Configuration;
+import org.apache.commons.configuration2.ConfigurationConverter;
+import org.apache.commons.configuration2.ImmutableConfiguration;
+import org.apache.commons.configuration2.MapConfiguration;
+import static org.apache.commons.lang3.Validate.notNull;
+
+import com.ocient.cli.ParseException;
+
+public class ExtractConfiguration {
+
+    ExtractConfiguration(final Properties properties){
+        notNull(properties, "ExtractConfiguration received null properties object");
+        final ImmutableConfiguration config = createConfig(properties);
+
+        // Only location type is required.
+        locationType = LocationType.valueOf(config.getString(LOCATION_TYPE).toUpperCase());
+        notNull(locationType, "Location type is a required configuration option");
+        // To achieve case insensitivity for file type
+        fileType = FileType.valueOf(config.getString(FILE_TYPE, DEFAULT_FILE_TYPE).toUpperCase());
```

Thoughts on using [EnumUtils.getEnumIgnoreCase](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/EnumUtils.html#getEnumIgnoreCase-java.lang.Class-java.lang.String-E-)?

---

No, the difference between that and `valueOf` is that instead of throwing, it resolves to a null. I think we want to just throw here so that this fails immediately.

---

As is, we're still using a String to represent an Enum for `DEFAULT_FILE_TYPE` (and `DEFAULT_COMPRESSION`).

If we change the type of `DEFAULT_FILE_TYPE` to `FileType`, we can use `getEnumIgnoreCase` with a `defaultEnum` argument. The code would look like:

```java
fileType = EnumUtils.getEnumIgnoreCase(FileType.class, config.getString(FILE_TYPE), DEFAULT_FILE_TYPE);
```

If you want to be consistent and use `getEnumIgnoreCase` for all of these, `notNull` returns the argument you pass it, so we can wrap the one used to get `locationType` with `notNull` like so:

```java
locationType  = notNull(EnumUtils.getEnumIgnoreCase(LocationType.class, config.getString(LOCATION_TYPE)), "Location type is a required configuration option");
```

but I'm also perfectly content sticking w/ `Enum.valueOf` for it, because there is no default.

---

Maybe I am misunderstanding you.
I think doing this for `fileType`:
`fileType = EnumUtils.getEnumIgnoreCase(FileType.class, config.getString(FILE_TYPE), DEFAULT_FILE_TYPE);
`
would result in getting the default (DELIMITED), if they ever specify something incorrect (`file_type="badThing"`), rather than erroring loudly which is what I want.

If I do it without the default argument,
`fileType = EnumUtils.getEnumIgnoreCase(FileType.class, config.getString(FILE_TYPE));`
then that would result in null if `file_type` is unspecified because `config.getString(FILE_TYPE)` would return null

---

I definitely think we want to catch something like `file_type="bad_thing"`. `EnumUtils.getEnumIgnoreCase` with a default hides that, so we shouldn't use it.

Lets stick with the code as is. However, we should catch the exceptions thrown by `Enum.valueOf` (looks like `IllegalArgumentException` and `NullPointerException`) and wrap them in a `ParseException` such that the user of this constructor only needs to catch `ParseException`.

---

done